book/src/assets/how_to_define_custom_assets.md

```diff
@@ -0,0 +1,279 @@
+# How to Define Custom Assets
+
+This guide explains how to define a new asset type to be used in an Amethyst application. If you are defining a new asset type that may be useful to others, please send us a PR!
+
+1. Define the type and handle for your asset.
+
+    ```rust,no_run,noplaypen
+    # extern crate amethyst;
+    # extern crate serde_derive;
+    #
+    use amethyst::{
+        assets::Handle,
+        ecs::VecStorage,
+    };
+
+    /// Custom asset representing an energy blast.
+    #[derive(Clone, Debug, Default, PartialEq, Eq)]
+    pub struct EnergyBlast {
```

Comment:
I think this is a bit of a poor example type. If using an example that's just multimedia is too restrictive, then I would suggest at least supplementing this example with some.

Reply:
Just to make sure I understand correctly:

* What do you mean by multimedia?
* Do you mean it's simpler with multimedia as an asset?

    Thought this was simpler &ndash; struct with 2 fields &ndash; unless you meant "this is too simple", and I should use an image / sound

Reply:
I meant that, as is, I think it's a little too simple. The restrictive remark was in regards to the potential concern that only using visuals/sounds in an Asset would be misleading as to the scope of an Asset. As is, however, I feel it could confuse people into thinking that that isn't what an Asset is for.

Reply:
> as is, I think it's a little too simple

So, from past work experience, the simple example is actually valuable:

* When it was made simple, people said "this is too simple, we want something closer to real world usage".
* When we gave them an example that was slightly more difficult, without the simple case, we hear back "there's too much of a conceptual leap, show us something easier first".

We couldn't "win", though the true win is actually the simple case &ndash; more people understood and learned &ndash; so we settle on that. Then if necessary we also added another more difficult example.

Also, if the opening page is read, then I think people can infer that "oh this `EnergyBlast` can be replaced with my audio / image type". Is that assumption alright? See second half of this comment.

---

> only using visuals/sounds in an Asset would be misleading as to the scope of an Asset

I believe the explanation on the [`assets`](https://github.com/amethyst/amethyst/blob/e5eed5b223af2ba179ac7523102c3d0bcff18686/book/src/assets.md) page should be enough for people to realize "it can be any *configuration* that your application needs". If the opening sentence isn't clear enough, I can include an example of [a serialized asset](https://github.com/amethyst/amethyst/pull/1251/files#diff-1909f34eb356799930dc82795b0d59c2R82), but I prefer people to not "use code to explain code", and rely on their understanding of "textures, sounds, and game level configuration".

Reply:
I'm not keen on relying on people making conclusion jumps. I can get your point on not "using code to explain code", so maybe a little extra wording to make the "it can be anything" point explicit?

Reply:
yeaps shall have a think!
also, maybe i'm just paranoid of the `.ron` file being "code" &ndash; trying to avoid the "let's explain `git` using terms like 'commits' and 'branches'" scenario.

Reply:
> a little extra wording to make the "it can be anything" point explicit

[here we go](https://github.com/amethyst/amethyst/compare/e5eed5b223af2ba179ac7523102c3d0bcff18686..2c58febb2ebca31aaff230fa94b2a500e475da58#diff-df2bed129120ea43546d75d912837f44)

Reply:
I agree with Kel's feedback on that.